[Aishwarya Hariharan Iyer] iP

Aishwarya-Hariharan-Iyer opened the pull request:
# Duke Aemon Of Old
> “You can do anything, but not everything.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

**Duke Aemon Of Old** :crown: has ~~broken out of~~ risen from the crypts _centuries_ after centuries of peace. No one knows why. Having come to the castle to investigate this mystery, you realize Aemon has excellent memory. As you fish for clues, use Duke Aemon to:

- [X] Jot down quick Todos
- [X] Pen events with dates and duration
- [X] Note important tasks with their deadlines
- [ ] Solve the puzzle of Aemon's return (Coming Soon!)

Run Duke by calling 


Here is how you can talk to Aemon:
1. Run Duke.java
2. Write a task description following `todo ` to save a Todo task
3. Write a task description and date with time (formatted as **description/yyyy-MM-dd HH:mm**) following `deadline ` to store a Deadline-based task
4. Write a task description and date with start and end times (formatted as **description/yyyy-MM-dd HH:mm HH:mm**) following `event ` to store an Event
5. Type `mark ` followed by the task number to mark a task as complete. Alternatively, use `unmark ` instead to mark it as incomplete
6. Type in `find ` followed by a keyword to see only the tasks containing the keyword
7. Type in `list ` to see all your saved tasks!

Play around with the main method to call other functions of Aemon like `echoAndExit` and learn more about your philosophical assistant born before Time.

```java
public static void main(String[] args) {
       Duke AemonT = new Duke();
       AemonT.interact();
    }
```
Why struggle with organizing a thousand tasks when you have a thousand-year-old friend in Aemon? With him, it is:
- **EASY** to plan your day
- **SIMPLE** to make changes
- **FUN** to live an organized life!

Commit:
Add Gradle support

Aishwarya-Hariharan-Iyer committed:
Change Duke to Duke Aemon of Old

Aishwarya-Hariharan-Iyer committed:
Add function of Duke to echo user input and exit

Aishwarya-Hariharan-Iyer committed:
Seperate features into distinct functions

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for methods created

Aishwarya-Hariharan-Iyer committed:
Add function for Duke to add item to list and display list

Aishwarya-Hariharan-Iyer committed:
Add class for list objects with task and status associated with each item

Aishwarya-Hariharan-Iyer committed:
Add makeList method to store and display task along with status

Aishwarya-Hariharan-Iyer committed:
Add function to change completion status of tasks in list

Aishwarya-Hariharan-Iyer committed:
Add seperate class for list object

Aishwarya-Hariharan-Iyer committed:
Add new class for ToDo tasks

Aishwarya-Hariharan-Iyer committed:
Add class for deadline based tasks

Aishwarya-Hariharan-Iyer committed:
Fix message displayed on adding deadline

Aishwarya-Hariharan-Iyer committed:
Add class for events along with existing ones for deadlines and todos

Aishwarya-Hariharan-Iyer committed:
Add function to add deadlines, events and todos to list

Aishwarya-Hariharan-Iyer committed:
Test using the I/O redirection technique

Aishwarya-Hariharan-Iyer committed:
Handle errors of empty input, invalid input and empty tasks

Aishwarya-Hariharan-Iyer committed:
Add function to delete items from list

Aishwarya-Hariharan-Iyer committed:
Add function to create file and write tasks to it

Aishwarya-Hariharan-Iyer committed:
Add function to save and retrieve list from Hard Disk

Aishwarya-Hariharan-Iyer committed:
Add ability to understand date in deadlines

Aishwarya-Hariharan-Iyer committed:
Add ability to understand dates in deadline

Aishwarya-Hariharan-Iyer committed:
Add ability to understand dates in deadline

Aishwarya-Hariharan-Iyer committed:
Add ability to understand date and time

Aishwarya-Hariharan-Iyer committed:
Revert "Add ability to understand dates in deadline"

This reverts commit d8773bc1678b641d1411af59acd4937609662f28.

Aishwarya-Hariharan-Iyer committed:
Revert "Revert "Add ability to understand dates in deadline""

This reverts commit 5dffbdf6dca816894ef730a79e70449c4f3a19ad.

Aishwarya-Hariharan-Iyer committed:
Merge pull request #1 from Aishwarya-Hariharan-Iyer/branch-Level-7

Branch level 7

Aishwarya-Hariharan-Iyer committed:
Resolve merge conflicts

Aishwarya-Hariharan-Iyer committed:
Merge pull request #2 from Aishwarya-Hariharan-Iyer/branch-Level-8

Branch level 8

Commit:
Add TaskList class to store Duke ListObjects

Aishwarya-Hariharan-Iyer committed:
Add TaskList class to store Duke ListObjects

Aishwarya-Hariharan-Iyer committed:
Add Storage class to edit and read task lists from hard disk

Aishwarya-Hariharan-Iyer committed:
Modify Storage to read and write to a file

Aishwarya-Hariharan-Iyer committed:
Add Ui class to read and respond to user input

Aishwarya-Hariharan-Iyer committed:
Add ParserDuke class to parse commands

Aishwarya-Hariharan-Iyer committed:
Modify Duke class to interact using other classes

Aishwarya-Hariharan-Iyer committed:
Categorize files into packages

Aishwarya-Hariharan-Iyer committed:
Merge pull request #3 from Aishwarya-Hariharan-Iyer/add-gradle-support

Add Gradle support

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Add Gradle support

Aishwarya-Hariharan-Iyer committed:
Fix errors in file creation and package relations

Aishwarya-Hariharan-Iyer committed:
Add 3 JUnit tests for basic testing of TaskList class

deepimpactmir commented:
Hi! I briefly looked through your code and it seems like you accidentally pushed some .class files into the remote repo.

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for Duke class

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for Deadline class

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for Event class

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for ToDo class

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for ListObject class

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for TaskList class

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for Ui class

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for Parser class

Aishwarya-Hariharan-Iyer committed:
Add JavaDoc for Storage class

Aishwarya-Hariharan-Iyer committed:
Merge pull request #4 from Aishwarya-Hariharan-Iyer/branch-A-JavaDoc

Merge branch-A-JavaDoc and master

Aishwarya-Hariharan-Iyer committed:
Change Event class to comply with coding conventions

Aishwarya-Hariharan-Iyer committed:
Move Event class to right folder

Aishwarya-Hariharan-Iyer committed:
Modify Duke class to follow coding conventions

Aishwarya-Hariharan-Iyer committed:
Modify ListObject, Deadline and ToDo classes to follow coding conventions

Aishwarya-Hariharan-Iyer committed:
Modify ParserDuke class to follow coding conventions

Aishwarya-Hariharan-Iyer committed:
Modify Storage class to follow coding conventions

Aishwarya-Hariharan-Iyer committed:
Modify TaskList class to follow coding conventions

Aishwarya-Hariharan-Iyer committed:
Modify Ui class to follow coding conventions

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Add JavaDoc for Storage class
  Add JavaDoc for Parser class
  Add JavaDoc for Ui class
  Add JavaDoc for TaskList class
  Add JavaDoc for ListObject class
  Add JavaDoc for ToDo class
  Add JavaDoc for Event class
  Add JavaDoc for Deadline class
  Add JavaDoc for Duke class

Aishwarya-Hariharan-Iyer committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Modify Ui class to follow coding conventions
  Modify TaskList class to follow coding conventions
  Modify Storage class to follow coding conventions
  Modify ParserDuke class to follow coding conventions
  Modify ListObject, Deadline and ToDo classes to follow coding conventions
  Modify Duke class to follow coding conventions
  Move Event class to right folder
  Change Event class to comply with coding conventions

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/listobjects/Deadline.java
#	src/main/java/duke/listobjects/Event.java
#	src/main/java/duke/listobjects/ListObject.java
#	src/main/java/duke/listobjects/ToDo.java
#	src/main/java/duke/parser/ParserDuke.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/tasklist/TaskList.java
#	src/main/java/duke/ui/Ui.java

Aishwarya-Hariharan-Iyer committed:
Add function to find tasks filtered by input keyword

deepimpactmir commented on build.gradle:
```diff
@@ -0,0 +1,41 @@
+}
+
+application {
+    mainClassName = "seedu.duke.Duke"
```
I think you forgot to change this, which should be `duke.Duke` for you.

deepimpactmir commented on src/main/java/duke/listobjects/Deadline.java:
```diff
@@ -0,0 +1,70 @@
+     * @return String representation of date and time in alternate format of MMM dd yyy HH:mm
+     */
+
+    public String formatDateTime(String txt) {
```
Looks like you have the same function in both Deadline, Event. Maybe it is better if you have a class that handles this for them instead?

Aishwarya-Hariharan-Iyer replied:
That is a good idea...thanks!

Aishwarya-Hariharan-Iyer replied:
Perhaps I could have a method in Tasks itself to differentially perform this task such that the absence of the input (as in todos) does not affect the way the String is parsed

deepimpactmir commented on src/main/java/duke/parser/ParserDuke.java:
```diff
@@ -0,0 +1,129 @@
+        } else {
+            TaskList listOfItems = Storage.readFromFile(FILE_PATH);
+
+            if (command.equals(LIST_CMD)) {
```
If else is fine but maybe switch is more readable?

Aishwarya-Hariharan-Iyer replied:
Yes, that would be better. Thanks!

Aishwarya-Hariharan-Iyer replied:
Maybe I could additionally employ enum and a separate class to improve the readability of the code...

deepimpactmir commented on src/main/java/duke/parser/ParserDuke.java:
```diff
@@ -0,0 +1,129 @@
+
+    private String command;
+
+    private final String LIST_CMD = "list";
```
Should these constants be static instead?

Aishwarya-Hariharan-Iyer replied:
Hmm...I didn't go with that approach though I considered it because for every command, I instantiate a ParserDuke (so there would be no static actions to perform). However, perhaps I can indeed improve memory space by making the entire process unveil in a static context.

deepimpactmir commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,70 @@
+            ArrayList<ListObject> listOfItems = lst.storeAllTasks();
+            //adapted from https://stackoverflow.com/questions/10404698/saving-arrays-to-the-hard-disk
+            ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(listFile));
+            out.writeObject(listOfItems);
```
I like how you used Serializable to write and read the tasks.

Aishwarya-Hariharan-Iyer replied:
Thanks!

deepimpactmir reviewed (commented):
I thought of using Serializable for saving/loading of tasks before and that is something I think is particularly clean. I also liked the personality that you given to the bot! :).

Aishwarya-Hariharan-Iyer committed:
Add function to filter tasks based on keyword

Aishwarya-Hariharan-Iyer committed:
Revert "Add function to filter tasks based on keyword"

This reverts commit 2df7cd1cebe5b850bc23f595c0cbb5364edc470d.

Aishwarya-Hariharan-Iyer committed:
Merge pull request #6 from Aishwarya-Hariharan-Iyer/branch-Level-9

Merge branch-Level-9 with master

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Add function to find tasks filtered by input keyword

tanxuanyun commented on src/main/java/duke/listobjects/Deadline.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+
+    /**
+     * Reads String representing date and time and returns an alternate String representation of it
+     * @param txt String representing date and time in the format of yyyy-MM-dd HH:mm
+     * @return String representation of date and time in alternate format of MMM dd yyy HH:mm
+     */
```
I like how your JavaDoc is very detailed and clear

Aishwarya-Hariharan-Iyer replied:
Thank you!

tanxuanyun commented on src/main/java/duke/parser/ParserDuke.java:
```diff
@@ -0,0 +1,134 @@
+                    System.out.println("Mark this on your calendar! \n" + newItem.toString()
+                            + "\nYou now have " + listOfItems.knowTaskCount() + " tasks to do!"
+                            + "\n***********************************************************************");
+                } else if (task.isEmpty()) {
+                    System.out.println("The folly of youth to cheat Time! Write your task following 'deadline'");
+                    System.out.println("***********************************************************************");
+                }
+            } else if (command.contains(EVENT_CMD)) {
+                String event1 = command.replaceAll("event ", "");
+                String[] words = event1.split("/");
+                String task = words[0];
+                String event = words[1];
+                if(!task.isEmpty()) {
+                    ListObject newItem = new Event(task, 0, event);
+                    listOfItems.handleItemAddition(newItem);
+                    Storage.makeListFile(FILE_PATH, listOfItems);
+                    System.out.println("Another moment to mark... \n" + newItem.toString()
+                            + "\nYou now have " + listOfItems.knowTaskCount() + " tasks to do!"
+                            + "\n***********************************************************************");
+                } else if (task.isEmpty()) {
+                    System.out.println("Come Alive! Write an activity following 'event'");
+                    System.out.println("***********************************************************************");
+                }
+            } else if (command.contains(FIND_CMD)) {
+                String target = command.replaceAll("find ", "");
+                System.out.println("Here is what you are looking for!");
+
+            } else {
+                System.out.println("Why trouble me with the unrefined language of the youth! Speak plainly, my friend!");
+                System.out.println("***********************************************************************");
+            }
```
Perhaps it may be better to abstract out some of the logic so that the function is not too long?

Aishwarya-Hariharan-Iyer replied:
Yes...perhaps a subclass of the Parser dealing with String manipulation on targets might help. Thanks for the suggestion!

tanxuanyun commented on src/main/java/duke/listobjects/ToDo.java:
```diff
@@ -0,0 +1,28 @@
+package duke.listobjects;
+
+
+/**
+ * Represents ToDo which is a listObject with only a task decription
+ */
+public class ToDo extends ListObject {
+
+    /**
+     * Constructs a ToDo object with given task description and status
+     * @param task String representing task description
+     * @param status int with value 1 if task is complete and 0 otherwise
+     */
+    public ToDo(String task, int status) {
+        super(task, status);
+    }
+
+
+    /**
+     * Returns String representing the ToDo object
+     * @return String representing ToDo object
+     */
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+
+}
```
I really like how this class is very succinct and has very detailed JavaDoc. Your code looks really neat here

Aishwarya-Hariharan-Iyer replied:
Thank you!

tanxuanyun commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,41 @@
+package duke;
```
I like how you extracted out the classes and packages, it makes the overall structure of your code really neat!

Aishwarya-Hariharan-Iyer replied:
Thanks! 😄

Aishwarya-Hariharan-Iyer reviewed (commented):
Reply to Review comments and suggestions

Aishwarya-Hariharan-Iyer reviewed (commented):
Responses to review comments and suggestions.

Aishwarya-Hariharan-Iyer committed:
Make skeleton for GUI

Aishwarya-Hariharan-Iyer committed:
Set up Duke GUI skeleton

Aishwarya-Hariharan-Iyer committed:
Create layout for user interaction

Aishwarya-Hariharan-Iyer committed:
Complete iteration 1 from Tutorial

Aishwarya-Hariharan-Iyer committed:
Complete Iteration 2 of tutorial

Aishwarya-Hariharan-Iyer committed:
Complete Final Iteration in Tutorial

Aishwarya-Hariharan-Iyer committed:
Add GUI for Duke

Aishwarya-Hariharan-Iyer committed:
Merge pull request #7 from Aishwarya-Hariharan-Iyer/Level-10

Merge Level-10 branch and master

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Add GUI for Duke
  Complete Final Iteration in Tutorial
  Complete Iteration 2 of tutorial
  Complete iteration 1 from Tutorial
  Create layout for user interaction
  Set up Duke GUI skeleton

Aishwarya-Hariharan-Iyer committed:
Modify Duke GUI responses

Aishwarya-Hariharan-Iyer committed:
Abstract out shared functions of Event and Deadline

Event and Deadline classes share time-based parameters and methods.

The common parameters and behavior cause code duplication.

Extracting the common behavior into the ListObject superclass
makes the codebase adhere to the DRY principle and scalable in future
to introduce other subclasses dealing with time associated with tasks.

Let's include the time field in the superclass ListObject and
put time reading and modification methods into the superclass.

Using such inheritance is preferable as it makes the codebase
more scalable should we wish to introduce new subclasses
that have a time field associated with the task.

Aishwarya-Hariharan-Iyer committed:
Remove if-else nests and replace them with switch statements

Parser Class contains three-layer deep if-else nests in method
responsible for parsing different user commands.

The deep nests cause arrowhead formation, make the code less readable
and do not adequately emphasize logical flow.

Let's use switch statements to handle different user commands.

Switch statements are preferable over if-else statements as they
prevent arrow-head formation, improve code readability, and
highlight the logical flow of conditions, thus improving code quality.

Aishwarya-Hariharan-Iyer committed:
Merge pull request #9 from Aishwarya-Hariharan-Iyer/branch-A-CodeQuality

Merge branch-A-CodeQuality with master

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Remove if-else nests and replace them with switch statements
  Abstract out shared functions of Event and Deadline

Aishwarya-Hariharan-Iyer committed:
Add Java assertions in various classes

Aishwarya-Hariharan-Iyer committed:
Merge pull request #10 from Aishwarya-Hariharan-Iyer/branch-A-Assertions

Merge branch-A-Assertions with master

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Add Java assertions in various classes

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip into branch-BCD-Extension

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Add Java assertions in various classes
  Remove if-else nests and replace them with switch statements
  Abstract out shared functions of Event and Deadline

Aishwarya-Hariharan-Iyer committed:
Implement Comparable interface in ListObject class

Aishwarya-Hariharan-Iyer committed:
Add sorting functionality in the TaskList class

Aishwarya-Hariharan-Iyer committed:
Merge pull request #11 from Aishwarya-Hariharan-Iyer/branch-BCD-Extension

Merge branch-BCD-Extension with master

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Add sorting functionality in the TaskList class
  Implement Comparable interface in ListObject class

Aishwarya-Hariharan-Iyer committed:
Improve GUI by changing images and alignment

Aishwarya-Hariharan-Iyer committed:
Add new images

Aishwarya-Hariharan-Iyer committed:
Update README.md with User Guide

Aishwarya-Hariharan-Iyer committed:
Update README.md as User Guide

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Update README.md as User Guide
  Update README.md with User Guide

Aishwarya-Hariharan-Iyer committed:
Add screenshot of current UI

Aishwarya-Hariharan-Iyer committed:
Add screenshot of current UI

Aishwarya-Hariharan-Iyer committed:
Place GUI-related files in new folder

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Add screenshot of current UI
  Add screenshot of current UI

Aishwarya-Hariharan-Iyer committed:
Fix bug in parsing commands and formatting response

Aishwarya-Hariharan-Iyer committed:
Modify exception handling to reflect stack trace

Aishwarya-Hariharan-Iyer committed:
Fix storage bug to account for empty files

Aishwarya-Hariharan-Iyer committed:
Modify design of TasksList's subclasses to account for more user cases

Aishwarya-Hariharan-Iyer committed:
Fix bug in compareTo method of ListObject to account for all fields

Aishwarya-Hariharan-Iyer committed:
Fix styling and formatting issues

Aishwarya-Hariharan-Iyer committed:
Add and handle EmptyCommandException

Aishwarya-Hariharan-Iyer committed:
Modify EmptyCommandExcpetion's message

Aishwarya-Hariharan-Iyer committed:
Separate parsing of tasks into separate method

Aishwarya-Hariharan-Iyer committed:
Add and handle more Exceptions

Aishwarya-Hariharan-Iyer committed:
Handle invalid item exceptions and separate parse commands

Aishwarya-Hariharan-Iyer committed:
Add personality to Duke Aemon and handle more errors

Aishwarya-Hariharan-Iyer committed:
Add function to display all commands to user

Aishwarya-Hariharan-Iyer committed:
Improve Javadocs by adding more details

Aishwarya-Hariharan-Iyer committed:
Use enums in ListObject class to customize methods

Aishwarya-Hariharan-Iyer committed:
Handle incorrect input date and time format and add JavaDocs

Aishwarya-Hariharan-Iyer committed:
Reformat code according to CheckStyle files

Aishwarya-Hariharan-Iyer committed:
Add more test cases and for more classes

Aishwarya-Hariharan-Iyer committed:
Update README.md with more details

Aishwarya-Hariharan-Iyer committed:
Fix formatting error

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Update README.md with more details

Aishwarya-Hariharan-Iyer committed:
Fix minor bug in parsing condition

Aishwarya-Hariharan-Iyer committed:
Update test case

Aishwarya-Hariharan-Iyer committed:
Update mainClassName in Gradle build

Aishwarya-Hariharan-Iyer committed:
Modify Storage to work on all systems

Aishwarya-Hariharan-Iyer committed:
Delete Ui.png

Aishwarya-Hariharan-Iyer committed:
Upload new Ui.png

Aishwarya-Hariharan-Iyer committed:
Add image to User Guide

Aishwarya-Hariharan-Iyer committed:
Add Set up instructions

Aishwarya-Hariharan-Iyer committed:
Update User Guide

Aishwarya-Hariharan-Iyer committed:
Merge branch 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip

* 'master' of https://github.com/Aishwarya-Hariharan-Iyer/ip:
  Update User Guide
  Add Set up instructions
  Add image to User Guide
  Upload new Ui.png
  Delete Ui.png

Aishwarya-Hariharan-Iyer committed:
Add documents and updated build properties

Aishwarya-Hariharan-Iyer committed:
Add developer and user guides

Aishwarya-Hariharan-Iyer committed:
Update tonal quality of exceptions

Aishwarya-Hariharan-Iyer committed:
Update Aemon's voice across functions and update tests

Aishwarya-Hariharan-Iyer committed:
Update the README details
